lib/breakfast_roulette/bot.ex

```diff
@@ -5,25 +5,45 @@ defmodule BreakfastRoulette.Bot do
     name: @bot
 
   alias BreakfastRoulette.Model.Person
+  alias BreakfastRoulette.Model.Group
 
   require Logger
 
   middleware(ExGram.Middleware.IgnoreUsername)
 
   def bot(), do: @bot
 
-  def handle({:command, "start", _msg}, context) do
-    answer(context, "Hi!")
+  def handle({:command, "start", msg}, context) do
+    tg_user = %{
+      first_name: msg.chat.first_name,
+      last_name: msg.chat.last_name,
+      telegram_id: to_string(msg.chat.id),
+      username: msg.chat.username
+    }
+
+    case Person.create_person(tg_user) do
+      {:ok, user} -> answer(context, "Hi #{user.first_name}!")
+      {:error, error} -> answer(context, error)
+    end
   end
 
-  def handle({:command, "register", %{from: tg_user}}, context) do
-    tg_user |> inspect |> Logger.debug()
+  def handle({:command, "register", msg}, context) do
+    msg |> inspect |> Logger.info()
 
-    tg_user = tg_user |> Map.put(:telegram_id, tg_user[:id] |> to_string) |> Map.delete(:id)
+    case msg.text do
```

Comment:
I think It would look cleaner if we match the empty string in a separate function clause:
```elixir
def handle({:command, "register", %{text: ""}}, context) do
  # Create group
end
```

Reply:
Yeah, that's better.

I'm not yet sure about the command names, maybe it's time to discuss. 

| Command              | Action                | Response                                 |
|----------------------|-----------------------|------------------------------------------|
| `/start`               | Registers the user    | Presents a list of groups?               |
| `/join [group_name]`   | Add a user to a group | Ok or If blank, presents a list of groups |
| `/create [group_name]` | Creates new group     | Ok or If blank, asks for a name           |

What do you think about?

Reply:
I think those commands are the ones! I agree on retrieving a list of groups on `/start`

Reply:
We should also include a `/leave` command.